ui/calendar.js

```diff
@@ -61,26 +61,27 @@ return $.widget( "ui.calendar", {
 		this.labels = Globalize.translate( "datepicker" );
 		this.buttonClickContext = this.element[ 0 ];
 
-		this.date = $.date( this.options.value, this.options.dateFormat ).select();
-		this.date.eachDay = this.options.eachDay;
+		this.date = $.date( this.options.value, this.options.dateFormat );
+		this.viewDate = this.date.clone();
+		this.viewDate.eachDay = this.options.eachDay;
 
 		this._on( this.element, {
 			"click .ui-calendar-prev": function( event ) {
 				event.preventDefault();
 				this.date.adjust( "M", -this.options.numberOfMonths );
-				this.refresh();
+				this._refresh();
```

Comment:
So `this.date` is just a temporary variable? And go away, if we go with "Shouldn't we pass date as a parameter to build* fns instead of setting this.date?"? Anyway, should't block this PR, can deal with that later.